src/components/hack-details.js

```diff
@@ -30,37 +34,72 @@ const useStyles = makeStyles(theme => ({
   },
   chipMArgin: {
     marginRight: theme.spacing(1)
+  },
+  marginTop: {
+    marginTop: theme.spacing(2)
+  },
+  marginFix: {
+    marginLeft: theme.spacing(0.1),
+    marginRight: theme.spacing(0.1),
+    paddingTop: theme.spacing(1),
+    paddingBottom: theme.spacing(1)
   }
 }));
 
 export function HackDetails(props) {
   const classes = useStyles();
 
-  const [data, setData] = useState(null);
+  const [displayData, setDsiplayData] = useState(null);
+  const [updateData, setUpdateData] = useState({});
+  const [edit, setEdit] = useState(false);
```

Comment:
I think this would be better driven through a route

`/hack-details/abc123`
`/hack-details/abc123/edit`

Reply:
This has been implemented along with redux